packages/freezed/lib/src/templates/copy_with.dart

```diff
@@ -197,10 +200,17 @@ $s''';
           data.makeCollectionsImmutable) {
         propertyName = '_$propertyName';
       }
-      var ternary =
-          '${p.name} == freezed ? $accessor.$propertyName : ${p.name} ';
+
+      var ternary = '${p.name} == freezed ? $accessor.$propertyName ';
+
+      final type = p.commonSubtype ?? p.type;
+      if (p.commonSubtype != null && p.type != p.commonSubtype) {
+        ternary += 'as $type ';
```

Comment:
I could be wrong, but isn't this still doing the invalid casts for copyWith for a common subtype of `num, int -> int`?

Are we moving the copyWith changes to another PR or are you still trying to do the `T & T? -> T` subtyping (which doesn't run into cast errors right?)

Reply:
This is only for the nullability subtyping which shouldn't run into problems at all.